homeassistant/components/azure_devops/__init__.py

```diff
@@ -0,0 +1,123 @@
+"""Support for Azure DevOps."""
+import logging
+from typing import Any, Dict
+
+from aioazuredevops.client import DevOpsClient
+import aiohttp
+
+from homeassistant.components.azure_devops.const import (
+    CONF_ORG,
+    CONF_PAT,
+    CONF_PROJECT,
+    DATA_AZURE_DEVOPS_CLIENT,
+    DATA_ORG,
+    DATA_PROJECT,
+    DOMAIN,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.typing import ConfigType, HomeAssistantType
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass: HomeAssistantType, config: ConfigType) -> bool:
+    """Set up the Azure DevOps components."""
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistantType, entry: ConfigEntry) -> bool:
+    """Set up Azure DevOps from a config entry."""
+    client = DevOpsClient()
+
+    try:
+        if entry.data[CONF_PAT] is not None:
+            await client.authorize(entry.data[CONF_PAT], entry.data[CONF_ORG])
+            if not client.authorized:
+                _LOGGER.warning(
+                    "Could not authorize with Azure DevOps. You may need to update your token."
+                )
+                raise ConfigEntryNotReady
+        await client.get_project(entry.data[CONF_ORG], entry.data[CONF_PROJECT])
+    except aiohttp.ClientError as exception:
+        _LOGGER.warning(exception)
+        raise ConfigEntryNotReady from exception
+
+    instance_key = f"{DOMAIN}_{entry.data[CONF_ORG]}_{entry.data[CONF_PROJECT]}"
+    hass.data.setdefault(instance_key, {})[DATA_AZURE_DEVOPS_CLIENT] = client
+    hass.data.setdefault(instance_key, {})[DATA_ORG] = entry.data[CONF_ORG]
+    hass.data.setdefault(instance_key, {})[DATA_PROJECT] = entry.data[CONF_PROJECT]
+
+    # Setup components
+    hass.async_create_task(
+        hass.config_entries.async_forward_entry_setup(entry, "sensor")
+    )
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistantType, entry: ConfigType) -> bool:
+    """Unload Azure DevOps config entry."""
+    # Unload sensors
+    await hass.config_entries.async_forward_entry_unload(entry, "sensor")
```

Comment:
This returns a boolean that we normally want to return from `async_unload_entry`.